src/test/java/net/datafaker/script/ProvidersDocsGenerator.java

```diff
@@ -0,0 +1,204 @@
+        }
+    }
+
+    private void writeProviderToTable(final String providerName, final String sinceTag) {
+        String inCompleteProviderEntry = "| " + "[" + addSpaceBetweenNameOfProvider(providerName) + "]" +
+            "({{ datafaker.javadoc }}/" + providerName + ".html)";
+        int lengthOfNameSection = inCompleteProviderEntry.length();
+        // Number of chars in 'Name' section, including first '|'.
+        // '|' in front of 'Description' section not included in length
+        final int totalLengthOfNameSection = 94;
+        int lengthOfNameSectionToFill = totalLengthOfNameSection - lengthOfNameSection;
+
+        final String completeNameSection = "\n" + inCompleteProviderEntry + Strings.repeat(" ", lengthOfNameSectionToFill) + "|";
+
+        final String providerDescriptionSectionEntry = Strings.repeat(" ", 132) ;
+
+        // Format of `@since tag` should be '#.#.#'
+        String providerSinceTagSectionEntry = "|" + " " + sinceTag + " " + "|";
+
+        try {
+            writeToFile(completeNameSection + providerDescriptionSectionEntry + providerSinceTagSectionEntry);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * When generating new file, the old one should be removed, because new data will be written in existing one.
+     *
+     * @param dataToWrite
+     * @throws IOException
+     */
+    private void writeToFile(String dataToWrite) throws IOException {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(destinationPlaceOfProvidersFile, true))) {
+            writer.write(dataToWrite);
+        }
+    }
+
+    /**
+     * Gets tag from JavaDoc comment
+     *
+     * @param javaDocComment            JavaDoc Comment which contains tag
+     * @return tag in format "#.#.#". # - number
+     */
+    private String matchSinceTag(String javaDocComment) {
+        Matcher comment = pattern.matcher(javaDocComment);
+
+        if (comment.find()) {
+            return comment.group();
+        }
+        return "";
+    }
+
+    /**
+     * Fills <code>TreeSet<code/> with providers as Class<?>
+     */
+    private void fillTreeSet() {
+        subTypes.addAll(reflections.get(SubTypes.of(AbstractProvider.class).asClass()));
+    }
+
+    private String addSpaceBetweenNameOfProvider(String providerName) {
+        if (providerName.isEmpty()) {
+            throw new IllegalArgumentException("Supplied provider's name is empty!");
+        }
+
+        // Do not add spaces in abbreviation
+        if (providersNameAbbreviation.contains(providerName)) {
+            return providerName;
+        }
+
+        StringBuilder providerBuilder = new StringBuilder();
+        providerBuilder.append(providerName.charAt(0));
+
+        for (int i = 1; i < providerName.length(); i++) {
+            if (Character.isUpperCase(providerName.charAt(i)) && providerName.charAt(i - 1) != ' ') {
```

Comment:
```suggestion
            if (Character.isUpperCase(providerName.charAt(i)) && Character.isLowerCase(providerName.charAt(i - 1))) {
```
then you probably don't need `providersNameAbbreviation` at all

Reply:
@snuyanzin Thank you for pointing at the problem